### PR TITLE
Feat: add tip for plugin editor

### PR DESCRIPTION
**Pull request by FangSen9000**
Please answer these questions before submitting a pull request, **or your PR will get closed**.

**Why submit this pull request?**

- [ ] Bugfix
- [x] New feature provided
- [x] Improve performance
- [ ] Backport patches

**What changes will this PR take into?**

We need to tell users clearly that which part of the data should be filled.
image

##Before the change:

##After the change:

**Related issues**

#1363

**Checklist:**

- [x] Did you explain what problem does this PR solve? Or what new features have been added?
- [x] Have you added corresponding test cases?
- [x] Have you modified the corresponding document?
- [x] Is this PR backward compatible? If it is not backward compatible, please discuss on the mailing list first

### CODE REVIEW

**Comment by FangSen9000** [comment]
@guoqqqi @Baoyuantop @bzp2010 @SkyeYoung 
This work has been completed. I think the degree of completion is very high. All the pages that need plug-ins have examples. Because it is non functional, there is no need for new test cases.

**Comment by FangSen9000** [comment]
WIP, do not merge yet

File: web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js [hunk 1]
```diff
@@ -94,7 +93,6 @@ context('Edit Service with Upstream', () => {
         force: true,
       })
       .should('value', data.ip1);
-    cy.get(selector.input).should('be.disabled');
```
Comment (FangSen9000) [inline web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js:97]: This is a relatively old test case. In any case, there will be no input box abandoned. It is meaningless to judge whether it is abandoned

File: web/cypress/e2e/rest/upstream-create_and_edit_upstream_with_no_nodes.cy.js [hunk 2]
```diff
@@ -64,7 +64,6 @@ context('Create and Delete Upstream', () => {
 
     cy.get(selector.upstreamNodeMinus0).should('not.exist');
     cy.contains('button', 'Next').should('not.be.disabled').click();
-    cy.contains('button', 'Next').should('not.be.disabled').click();
```
Comment (FangSen9000) [inline web/cypress/e2e/rest/upstream-create_and_edit_upstream_with_no_nodes.cy.js:67]: There is an extra next added here. In fact, only one next is needed

File: web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js (hunk 1 above)
Comment (FangSen9000) [inline web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js:97]: This is a relatively old test case. In any case, there will be no input box abandoned. It is meaningless to judge whether it is abandoned

File: web/cypress/e2e/plugin/create-delete-in-drawer-plugin.cy.js [hunk 3]
```diff
@@ -292,12 +292,12 @@ context('Delete Plugin List with the Drawer', () => {
...
         .parents(selector.pluginCardBordered)
         .within(() => {
-          cy.get('button').click();
+          cy.get('button', 'Enable').click();
         });
       cy.get(selector.drawer)
         .should('be.visible')
         .within(() => {
-          cy.get(selector.checkedSwitcher).should('exist');
+          cy.get(selector.disabledSwitcher).should('exist');
```
Comment (FangSen9000) [inline web/cypress/e2e/plugin/create-delete-in-drawer-plugin.cy.js:300]: When the plug-in is enabled, the plug-in is not selected by default

File: web/cypress/e2e/plugin/create-route-with-plugin-orchestration.cy.js [hunk 4]
```diff
@@ -59,6 +59,7 @@ context('Create and delete route with plugin orchestration', () => {
     cy.contains('Next').click();
 
     cy.get(selector.groupButton).contains('Orchestration').click();
+    cy.wait(1000);
```
Comment (FangSen9000) [inline web/cypress/e2e/plugin/create-route-with-plugin-orchestration.cy.js:62]: This can reduce the retry time of cypress by about 2 times.

File: web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js [hunk 5]
```diff
@@ -73,7 +73,7 @@ context('Edit Service with Upstream', () => {
     cy.get(selector.description).type(data.description);
     cy.get(selector.upstreamSelector).click();
     cy.contains(data.upstreamName).click();
-    cy.get(selector.input).should('be.disabled');
+    cy.get(selector.input).should('not.be.disabled');
```
Comment (FangSen9000) [inline web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js:76]: name has just been written. It will not be abandoned until I click ‘next’

File: web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js [hunk 6]
```diff
@@ -73,7 +73,6 @@ context('Edit Service with Upstream', () => {
     cy.get(selector.description).type(data.description);
     cy.get(selector.upstreamSelector).click();
     cy.contains(data.upstreamName).click();
-    cy.get(selector.input).should('be.disabled');
```
Comment (FangSen9000) [inline web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js:76]: If you don't click next, some will be banned and some will not. Its positioning of elements is not accurate. It may refer to all input boxes, but there are forbidden and good input boxes. No matter whether it is '.not ' or '.be', it will make mistakes. If it refers to the box just entered, there is something wrong with its writing. It's better to delete it.

File: web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js [hunk 7]
```diff
@@ -88,19 +87,16 @@ context('Edit Service with Upstream', () => {
...
-
     cy.get(selector.upstreamSelector).click();
-    cy.contains('.ant-select-item-option-content', 'Custom').click();
-    cy.get(selector.nodes_0_host).should('not.be.disabled').clear().type(data.ip2);
-    cy.get(selector.nodes_0_port).type(data.port);
-    cy.get(selector.nodes_0_weight).type(data.weight);
+    cy.wait(500);
+    cy.contains('.ant-select-item-option-content', 'Custom').should('be.visible').click();
+    cy.wait(300);
+    cy.get$(selector.nodes_0_host).should('not.be.disabled').clear().type(data.ip2);
```
Comment (FangSen9000) [inline web/cypress/e2e/rest/service-edit-service-with-upstream.cy.js:94]: I use jQuery to wrap some elements, which can make the positioning of more accurate

**Comment by FangSen9000** [comment]
@guoqqqi, you said before that you would consider merging only if E2E is completely correct, but recently you should have found that the current test is problematic, but it is completely correct and needs to be fixed. Among them, five test cases are problematic, and they are difficult to pass. (this is not my problem, but the new E2E test case error that occurred after ten days ago) are: 
1. _Create delete in drawer plugin.cy.js_ 
2. _Upstream create_ and_ edit_ upstream_ with_ no_ Nodes.cy.js_ 
3. _Service-edit-service-with-upstream.cy.js_ (this mainly occurs in the cloud) 
4. _Plugin-schema.cy.js_.

Fortunately, I solved the first and second. The fourth one is the most serious one. It can not pass at all. It is a loop test. There is a problem with a judgment in the middle. There is no problem in the test logic, but somehow the judgment fails and the loop jumps out midway. You can see this https://github.com/apache/apisix-dashboard/pull/2612/checks. If there is no change, there will be 100% errors. I tried to spend three days, but I didn't find the root cause. (the third and fourth should find a more professional tester to have a look. The pass rate of the third is only 10% in a certain environment. The elements are either separated, lost or missing, and only appear in the cloud. It is normal locally, as shown in the video below:)

https://user-images.githubusercontent.com/72308243/189047162-0da14fc5-a3fa-47ee-8ab4-821647624c8f.mp4

> @Baoyuantop @bzp2010 If you have time, you can give some suggestions. And, If anyone can try to fix the test cases of 3 and 4, you can find them. Many people have these two problems in PR. 

To sum up, the **effect** of this PR is: add tip and preset model for plugin editor, improve E2E stability
All my work on improving test stability can be seen in this pr. This should reduce the error rate of the test.
For the explanation of model, please see the text at the top of this pr. It helps users to use the plug-in by presetting the plug-in template. There is a video.

File: web/cypress/e2e/plugin/create-delete-in-drawer-plugin.cy.js [hunk 8]
```diff
@@ -292,12 +292,12 @@ context('Delete Plugin List with the Drawer', () => {
       cy.contains(item)
         .parents(selector.pluginCardBordered)
         .within(() => {
-          cy.get('button').click();
+          cy.get('button', 'Enable').click();
```
[the remark this answers is not in the document]
Comment (FangSen9000) [reply web/cypress/e2e/plugin/create-delete-in-drawer-plugin.cy.js:295]: Well, I thought this would be more stable, and I would cancel the 'enable'. 

> As for "this E2E test actually disabled the plugins and then attempted to check if the plugin was enabled (or switch is in" on "state) which is causing failure in CI." 

what do you think should be solved? You can change it directly in my PR
@butterfly1924 
cc @bzp2010

File: web/src/components/Plugin/PluginDetail.tsx [hunk 9]
```diff
@@ -43,6 +48,8 @@ import type { languages } from 'monaco-editor';
 import { fetchSchema } from './service';
 import { json2yaml, yaml2json } from '../../helpers';
 import { PluginForm, PLUGIN_UI_LIST } from './UI';
+import*as allModels from './Models';
+import*as modelCode from './modelCode';
```
Comment (SkyeYoung) [inline web/src/components/Plugin/PluginDetail.tsx:52]:
```suggestion
import * as allModels from './Models';
import * as modelCode from './modelCode';
```

**Review by SkyeYoung — APPROVED** [review]

**Comment by FangSen9000** [comment]
@Baoyuantop @guoqqqi @bzp2010 @SkyeYoung 

All test cases of Dashboard have been repaired and passed, and the issue#2615 can be closed.

image
image

cc @butterfly1924 **plugin-schema.cy.js** has been fixed.

File: web/cypress/e2e/plugin/plugin-schema.cy.js [hunk 10]
```diff
@@ -75,10 +75,6 @@ describe('Plugin Schema Test', () => {
             } else {
               cy.log(`${name} not a global plugin, skipping`);
             }
-
-            if (cases.length === i + 1) {
-              cy.reload(true);
-            }
```
Comment (FangSen9000) [inline web/cypress/e2e/plugin/plugin-schema.cy.js:81]: Frequent refreshing is the root cause of plugin-schema.cy.js' error. In fact, the logic of the test case is completely correct. It is the constant aspiration of **umi.js**

**Comment by SkyeYoung** [comment]
cc @Baoyuantop @bzp2010

**Review by LiteSun — APPROVED** [review]

**Comment by FangSen9000** [comment]
I've removed the prompt box, maybe it'll be a little bit cleaner @SkyeYoung 
image
cc @Baoyuantop @guoqqqi

**Review by Baoyuantop — APPROVED** [review]

**Review by guoqqqi — APPROVED** [review]
